### PR TITLE
feat(node): v16.6

### PR DESCRIPTION
**Pull request by SimonSchick**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V16.md#16.6.0
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### Notes

- v16.5 introduced code to remove double examples, we generally prefer the es6 module style, this deletes *A LOT* of docs.
- v16.5 Set new default eslib to 2020
- v16.6 Removes ES string functions as they have been stable in default libs for a while.
- v16.6 Moved `inspector` and `trace_events` tests to their own files.

### CODE REVIEW

File: types/node/scripts/generate-docs/html-doc-processing.ts
Comment (amcasey) [inline types/node/scripts/generate-docs/html-doc-processing.ts:104]: Typo: "definitely"

**Review by amcasey — COMMENTED** [review]
LGTM, but I'd like to hear from another owner.

File: types/node/scripts/generate-docs/html-doc-processing.ts
Comment (SimonSchick) [reply types/node/scripts/generate-docs/html-doc-processing.ts:104]: One would think I'd know how to spell this by now...

**Review by peterblazejewicz — APPROVED** [review]
@SimonSchick LGTM, merge at your discretion once you type definitely

**Comment by SimonSchick** [comment]
@peterblazejewicz

Can we make the bot less aggressive so that single character changes in dt files may not require re-approval?

**Comment by amcasey** [comment]
@SimonSchick Once a package is marked Critical, basically any change requires maintainer review.

**Comment by amcasey** [comment]
@SimonSchick What's happening with the `at` debate?  Adding some, but not all, of the signatures with this PR seems like a recipe for confusion.

**Comment by SimonSchick** [comment]
Should I pull in all then?

**Comment by amcasey** [comment]
Yeah, I'd lean towards declaring them all directly, rather than pulling in another package or declaring only some of them.

**Review by amcasey — APPROVED** [review]

**Comment by SimonSchick** [comment]
Ready to merge

**Pull request merged**
